### DISCUSSION TITLE
Custom theme for email not working

### DISCUSSION THREAD

**Discussion by Nickk4** [keycloak/keycloak#22495] [Q&A] [answered]
I'm new to KeyCloak (21.0.1) and want to style the emails. I created the folders/files below and selected the custom theme under Realm settings.

themes/mytheme/email/theme.properties
```
parent=base
import=common/keycloak
styles=style.css
```

themes/mytheme/email/html/template.ftl
```
<#macro emailLayout>
<html>
<body>
    <img src="https://mysite.com/assets/logo.png" alt="myLogo" style="width:120px;height:44px;">
    <#nested>
</body>
</html>
</#macro>
```

themes/mytheme/email/resources/style.css
```
@import url('https://fonts.googleapis.com/css2?family=Roboto:wght@100;300;400;500;700;900&display=swap');
img {
    position: relative;
    margin: 0 auto 2rem auto;
}
p {
    font-family: "Roboto",sans-serif;
    font-size: 1rem;
}
```

The emails it now sends out indeed have the image added above the text. However, the styling is not applied (in MS Outlook 2021):
- The inline style for width and height is not applied. When I look at the email source code, it is there though.
- None of the styling from style.css is applied.

Any idea what I might be doing wrong?

**Comment by justin-tay** [comment, marked as answer]
For the image you can try

```
<img src="https://mysite.com/assets/logo.png" alt="myLogo" width="120" height="44">
```

For the rest of the styling it's hard to say. It looks like from Outlook 2007 onwards it is using Microsoft Word to render HTML and CSS which seems to only support a subset. You might want to try with a different e-mail client and if that works it's probably due to how Outlook does it.

**Reply by Nickk4** [reply]
Yes, this indeed works. Thanks!
